vermouth/utils.py

```diff
@@ -127,3 +141,53 @@ def are_all_equal(iterable):
     iterator = iter(iterable)
     first = next(iterator, None)
     return all(item == first for item in iterator)
+
+
+def are_different(left, right):
+    """
+    Return True if two values are different from one another.
+
+    Values are considered different if they do not share the same type. In case
+    of numerical value, the comparison is done with :func:`numpy.isclose` to
+    account for rounding. In the context of this test, `nan` compares equal to
+    itself, which is not the default behavior.
+    """
+    if left.__class__ != right.__class__:
+        return True
+    
+    if left is None:
+        return False
```

Comment:
Why? Can't I compare None and None?

Reply:
if `left` is `None`, then `right` is `None` as well (because the test before checked they were the same type) so they are not different. I have this case early because it is a common one.

Reply:
Could you add a comment?